### PR TITLE
[PYTX] VPDQ test utils

### PR DESCRIPTION
**Pull request by tigerchensh**
Summary
---------
Add utils for creating random pdqHash, similar pdqHash and a list of random VPDQ features for future tests and benchmarks.

Test Plan
---------

pytest
Run similar_hash for ten times and check if it meets the expected distance.
Check raise_error on invalid distance(256).
Create vpdq signal string and validate.

### CODE REVIEW

File: python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py [hunk 1]
```diff
@@ -35,6 +38,33 @@
...
+        random_dist = random.randint(0, 255)
+        random_hash = get_random_hash()
+        similar_hash = get_similar_hash(random_hash, random_dist)
+        assert simple_distance(similar_hash, random_hash) == random_dist
+    try:
+        invalid_dist = 256
+        get_similar_hash(get_random_hash(), invalid_dist)
+        assert False
+    except ValueError:
+        assert True
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py:56]: blocking: use the pytest helper for this.

  https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions

**Review by Dcallies — CHANGES_REQUESTED** [review]
Thanks for doing this! Comments inline

tl;dr:
1. A bunch of this is PDQ testing utils, not vPDQ, move them to live with those
2. Maybe some correctness bugs

File: python-threatexchange/threatexchange/extensions/vpdq/video_vpdq.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def validate_signal_str(cls, signal_str: str) -> str:
         vpdq_hashes = json_to_vpdq(signal_str)
         last_frame_number = -1
         for hash in vpdq_hashes:
-            if not re.match("^[0-9a-f]{64}$", hash.hex):
+            if not re.match("^[0-9a-f]{64}$", str(hash.hex, "utf-8")):
```
[the remark this answers is not in the document]
Comment (tigerchensh) [reply python-threatexchange/threatexchange/extensions/vpdq/video_vpdq.py:66]: It is because the hex_str created by the vpdq package is actually c_string. It needs to be decoded first and then uses re-match. I wasn't sure where I should implement this decode, because all the hash/distancing in vpdq are done in C++ library. But for easier usage, only having c_string as an internal usage within vpdq package would be better. Any suggestions on this?

File: python-threatexchange/threatexchange/extensions/vpdq/vpdq_util.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def json_to_vpdq(json_str: str) -> t.List[vpdq.VpdqFeature]:
     for frame_number, feature in vpdq_json.items():
         features.append(
             vpdq.VpdqFeature(
-                feature[QUALITY], frame_number, feature[HASH], feature[TIMESTAMP]
+                feature[QUALITY], int(frame_number), feature[HASH], feature[TIMESTAMP]
```
[the remark this answers is not in the document]
Comment (tigerchensh) [reply python-threatexchange/threatexchange/extensions/vpdq/vpdq_util.py:51]: It is because json object key is automatically converted to string (javascript only accepts string as key). That's why a cast is needed when loading the jsons with frame_number as the key.

**Comment by tigerchensh** [comment]
Thanks for the review. Since there are lots of utils written here are for PDQ and there is a problem with VPDQ package. I will close this PR and separate those changes into different PRs.
